pyproject.toml

```diff
@@ -101,73 +101,73 @@ classifiers = [
 [tool.poetry.dependencies]
 python = "^3.9,!=3.9.7,<3.12"
 
-aenum = "^3.0"
-aiohttp = "^3.8.1"
-backports-datetime-fromisoformat = "^2.0.0"
-beautifulsoup4 = "^4.9"
-cachetools = "^5.2"
-click = "^8.0"
-click-params = "^0.4.1"
-cloup = "^3.0.1"
-deprecation = "^2.1"
-diskcache = "^5.4.0"
-environs = "^9.4.0"
+aenum = ">=3,<3.2"
+aiohttp = ">=3.8,<3.9"
+backports-datetime-fromisoformat = ">=2,<3"
+beautifulsoup4 = ">=4.9,<5"
+cachetools = ">=5.2,<6"
+click = ">=8,<9"
+click-params = ">=0.4.1,<5"
+cloup = ">=3.0.1,<4"
+deprecation = ">=2.1,<3"
+diskcache = ">=5.4.0,<6"
+environs = ">=9.4.0,<10"
 fsspec = "^2023.01"
-lxml = "^4.9.1"
-measurement = "^3.2"
-numpy = "^1.22"
-pandas = "^2.0"
-Pint = "^0.17"
-platformdirs = "^3.8"
+lxml = ">=4.9.1,<5"
+measurement = ">=3.2,<4"
+numpy = ">=1.22,<1.24"
+pandas = ">=2,<2.1"
+Pint = ">=0.17,<0.23"
```

Comment:
Also here: If that technically works, because the package is compatible enough, I believe it is the right way to write down dependencies in ranges of version numbers, instead of anything else - modulo exceptions ;].